### PR TITLE
[ptftest] Add p4_dash_utils

### PR DESCRIPTION
**Pull request by jimmyzhai**
This PR adds a module p4_dash_utils to include helper functions for p4 pipeline. 
* Function `set_internal_config` aims to update pipeline internal configuration, neighbor mac, data port mac, cpu port mac, etc.
* Decorator `use_flow` aims to enable flow lookup stage in p4 pipeline. It wrappers method setUp/tearDown of test class.

### CODE REVIEW

**Comment by KrisNey-MSFT** [comment]
Author used P4 runtime directly to add a fixture to add underlay routing
PTF not using .py tests, so we will add more of a description and example
fyi SAI Challenger uses .py test

File: test/test-cases/functional/ptf/p4_dash_utils.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    req = p4runtime_pb2.WriteRequest()
+    req.device_id = 0
+    update = req.updates.add()
+    update.type = p4runtime_pb2.Update.INSERT
+    update.entity.table_entry.CopyFrom(entry)
+    stub.Write(req)
+
+
+def use_flow(cls):
```
Comment (r12f) [inline test/test-cases/functional/ptf/p4_dash_utils.py:94]: better make it more generic.

File: test/test-cases/functional/ptf/p4_dash_utils.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
+import grpc
```
Comment (r12f) [inline test/test-cases/functional/ptf/p4_dash_utils.py:1]: better add 1 example to show how to use it.

Comment (jimmyzhai) [reply test/test-cases/functional/ptf/p4_dash_utils.py:1]: Demo the use of `set_internal_config` in `saidashvnet_sanity.py`.

File: test/test-cases/functional/ptf/p4_dash_utils.py (hunk 1 above)
Comment (jimmyzhai) [reply test/test-cases/functional/ptf/p4_dash_utils.py:155]: Add general function `set_internal_config`.

**Review by r12f — APPROVED** [review]
lgtm!

File: dash-pipeline/bmv2/dash_metadata.p4 [hunk 3]
```diff
@@ -243,6 +243,7 @@ struct metadata_t {
     bit<16> dash_tunnel_next_hop_id;
     bit<32> meter_class;
     bit<8> local_region_id;
+    EthernetAddress cpu_mac;
```
Comment (r12f) [inline dash-pipeline/bmv2/dash_metadata.p4:246]: dpapp_mac

File: test/test-cases/functional/ptf/p4_dash_utils.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        for action in self.config.actions:
+            if action.preamble.name == name:
+                return action
+
+        return None
+
+
+def set_internal_config(neighbor_mac :bytes = None,
+                        mac :bytes = None,
+                        dpapp_mac :bytes = None,
```
Comment (r12f) [inline test/test-cases/functional/ptf/p4_dash_utils.py:50]: cpu_mac

File: test/test-cases/functional/ptf/saidashvnet_sanity.py [hunk 5]
```diff
@@ -25,6 +25,11 @@ def setUp(self):
         # SAI address family
         self.sai_ip_addr_family = SAI_IP_ADDR_FAMILY_IPV4
 
+        self.dut_mac   = get_mac("veth0")
```
Comment (r12f) [inline test/test-cases/functional/ptf/saidashvnet_sanity.py:28]: nit: space.

Comment (jimmyzhai) [reply test/test-cases/functional/ptf/saidashvnet_sanity.py]: fixed

File: test/test-cases/functional/ptf/p4_dash_utils.py (hunk 4 above)
Comment (jimmyzhai) [reply test/test-cases/functional/ptf/p4_dash_utils.py]: fixed

**Review by r12f — APPROVED** [review]
lgtm!

**Pull request merged**